laurent22/joplin
# All: Resolves #8466: WebDAV: Show a more descriptive error message when the password is empty

# Summary
This pull request makes two changes:
1. If the WebDAV error has code [401 (unauthorized)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) or [403 (forbidden)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403) and the password field is empty, the error message now tells the user to re-enter the WebDAV password.
2. If the error has code 401, the request is no longer retried. Previously, such an error would cause the API to retry the request several times. As a result, it could take some time (roughly 10 seconds) for sync to complete and the error message to be displayed.

This should fix #8466 if the server produces a 401 or 403 error when the user's password is empty (as Apache does).

# Screenshots

<img width="230" alt="Screenshot: You may have to re-enter your credentials above the sync button." src="https://github.com/laurent22/joplin/assets/46334387/a6b8d3cc-4fe5-4033-a2e9-1427d985d079">


# Testing
1. Set up a WebDAV sync target
2. Quit Joplin
4. Delete the WebDAV password from the system keychain (should end with "sync.6.password")
5. Re-open Joplin

This has only been tested on MacOS.
